api/api.js

```diff
@@ -0,0 +1,101 @@
+const request = require('request');
+const express = require('express');
+
+// MongoDB credentials setup
+const MongoClient = require('mongodb').MongoClient;
+const { password } = require('../password');
+const uri = `mongodb+srv://admin:${password}@course-plan-t2nrj.mongodb.net/test?retryWrites=true&w=majority`;
+const client = new MongoClient(uri, { useNewUrlParser: true, useUnifiedTopology: true });
+
+
+function getCourses(ros, callback) {
+    // Function to get all courses from specific roster
+
+    let result = [];
+    getSubjects("FA19", subjects => {
+
+        let subFill = [];
+        subjects.map((sub) => {
+            
+            request(`https://classes.cornell.edu/api/2.0/search/classes.json?roster=${ros}&subject=${sub}`, { json: true }, (err, res, body) => {
+                if (body) {
+                    subFill.push(sub);
+                    courseFill = [];
+                    let courses = body.data.classes;
+
+                    courses.map((course) => {
+                        result.push(course);
+                        courseFill.push(course);
+                        if (subFill.length === subjects.length && courseFill.length === courses.length) {
+                            return callback(result);
+                        }
+                    })
+                }
+            });
+        });
+    });
+}
+
+function getSubjects(ros, callback) {
```

Comment:
Using `async/await` may make the code more readable.
Also, consider adding jsdoc to the parameters to make document its usage, especially for the function that involves callbacks. Or even better, use typescript :)